src/lifecycles/mount.js

```diff
@@ -30,14 +31,23 @@ export function toMountPromise(appOrParcel, hardFail = false) {
         return appOrParcel;
       })
       .catch(err => {
-        if (!hardFail) {
-          handleAppError(err, appOrParcel);
-          appOrParcel.status = SKIP_BECAUSE_BROKEN;
-          return appOrParcel;
-        } else {
-          const transformedErr = transformErr(err, appOrParcel)
-          appOrParcel.status = SKIP_BECAUSE_BROKEN;
-          throw transformedErr
+        // If we fail to mount the appOrParcel, we should attempt to unmount it before putting in SKIP_BECAUSE_BROKEN
+        // We temporarily put the appOrParcel into MOUNTED status so that toUnmountPromise actually attempts to unmount it
+        // instead of just doing a no-op.
+        appOrParcel.status = MOUNTED
+        return toUnmountPromise(appOrParcel)
+          .then(setSkipBecauseBroken, setSkipBecauseBroken)
```

Comment:
We only want to call setSkipBecauseBroken once, and we want to do it regardless of whether the unmount succeeds or not. So a `.then(fn, fn)` is better than a `.then(fn).catch(fn)` in this case